spec/mongoid/traversable_spec.rb

```diff
@@ -251,4 +251,87 @@
       end
     end
   end
+
+  describe "#discriminator_key" do
+
+    context "when the discriminator key is not changed" do 
+      it "equals _type" do
+        expect(Instrument.discriminator_key).to eq("_type")
+      end
+
+      it "children's discriminator keys equal _type" do
+        expect(Guitar.discriminator_key).to eq("dkey")
+        expect(Piano.discriminator_key).to eq("_type")
+      end
+
+      it "the base discriminator key is _type" do
+        expect(Mongoid.discriminator_key).to eq("_type")
+      end
+    end
+
+    context "when the discriminator key is changed at the base level" do 
+      before do
+        Mongoid.discriminator_key = "hello"
+      end
+
+      after do
+        Mongoid.discriminator_key = "_type"
+      end
+
+      it "the value changes" do 
+        expect(Mongoid.discriminator_key).to eq("hello")
+      end
+
+      it "all discriminator keys change" do 
+        expect(Instrument.discriminator_key).to eq("hello")
+        expect(Guitar.discriminator_key).to eq("dkey")
+        expect(Piano.discriminator_key).to eq("hello")
+      end
+    end
+
+    context "when the discriminator key is changed in the parent" do 
+      before do
+        Instrument.discriminator_key = "hello2"
+      end
+
+      after do 
+        Instrument.discriminator_key = "_type"
+      end
+
+      it "changes in the child classes" do 
```

Comment:
Let's make it so that discriminator_key cannot be changed midway in the inheritance hierarchy, like store_in works.